cobc/codegen.c

```diff
@@ -9271,6 +9271,40 @@ output_key_components (struct cb_file* f, struct cb_key_component* key_component
 	}
 }
 
+
+static void
+output_indexed_file_key_colseq (const struct cb_file *f, const struct cb_alt_key *ak, int idx)
+{
+	cb_tree	key = ak ? ak->key : f->key;
+	cb_tree	key_col = ak ? ak->collating_sequence_key : f->collating_sequence_key;
+	int	type = cb_tree_type (key, cb_code_field (key));
+	cb_tree	col = NULL;
+
+	/* We only apply a collating sequence if the key is alphanumeric / display */
+	if ((type & COB_TYPE_ALNUM) || (type == COB_TYPE_NUMERIC_DISPLAY)) {
+		col = key_col ? key_col : f->collating_sequence;
+#if 0
+	/* TODO: this should be done for national, when available */
+	} else if (type & COB_TYPE_NATIONAL) {
+		col = key_col_n ? key_col_n : f->collating_sequence_n;
+#endif
+	}
+
+	output_prefix ();
+	if (idx == 0) {
+		output ("%s%s->collating_sequence = ", CB_PREFIX_KEYS, f->cname);
+	} else {
+		output ("(%s%s + %d)->collating_sequence = ", CB_PREFIX_KEYS, f->cname, idx);
+	}
+	if ((col != NULL) && CB_REFERENCE_P (col)) {
```

Comment:
Not necessarily a change, more a question - is there a reason to wrap the first part in parens?

Reply:
Ah, that's just my over-parenthesing OCD kicking-in. I'll drop them.